GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -652,6 +652,36 @@ public void remove() {
             throw new UnsupportedOperationException();
         }
     }
+
+    /**
+     * Add {@code childComponent} as a child of this object (owner object of the
+     * component is added as child). Adding a component will increase the
+     * {@link getChildrenCount() getChildrenCount()} for this scene object.
+     * 
+     * @param childComponet
+     *            {@link GVRComponent Component} to add as a child of this
+     *            object.
+     */
+    public void addChildComponent(GVRComponent childComponent) {
```

Comment:
It looks good to me now. I am also okay with overloading to addChildObject(GVRComponent com), instead of new term of “xxxComponent”.